Sources/Amplitude/DefaultTrackingOptions.swift

```diff
@@ -8,20 +9,38 @@ public class DefaultTrackingOptions {
         DefaultTrackingOptions(sessions: false, appLifecycles: false, screenViews: false)
     }
 
-    public var sessions: Bool = true
-    public var appLifecycles: Bool
-    public var screenViews: Bool
-    public var userInteractions: Bool
+    public var sessions: Bool {
+        didSet {
+            autocapture?.sessions = sessions
+        }
+    }
+
+    public var appLifecycles: Bool {
+        didSet {
+            autocapture?.sessions = appLifecycles
+        }
+    }
+
+    public var screenViews: Bool {
+        didSet {
+            autocapture?.sessions = screenViews
+        }
+    }
 
     public init(
         sessions: Bool = true,
         appLifecycles: Bool = false,
-        screenViews: Bool = false,
-        userInteractions: Bool = false
+        screenViews: Bool = false
     ) {
         self.sessions = sessions
         self.appLifecycles = appLifecycles
         self.screenViews = screenViews
-        self.userInteractions = userInteractions
+    }
+
+    private var autocapture: AutocaptureOptions?
+
+    func withAutocptureOptions(_ autocapture: AutocaptureOptions) -> DefaultTrackingOptions {
```

Comment:
Is there currently any way this can change other than the user setting a new autcapture options? I'm perfectly fine not handling edge cases that only occur when the user already has to modify their code (this is relatively noisy - easily caught by minimal testing of the feature they are changing), and generally would prefer the simplest workable solution.

If you're trying to go to these lengths to maintain compatibility, please add a test or at least a comment so it's obvious if someone tries to modify the code.

Reply:
I changed the approach to simplify the logic, with the tradeoff of not tracking changes to defaultTracking options after initialization. Please see f68eb0c0d991c20595c11072680eb8889fe0afe4.